main.tf

```diff
@@ -0,0 +1,46 @@
+resource "aws_autoscaling_group" "app" {
+  vpc_zone_identifier  = ["${var.subnet_list}"]
+  min_size             = "${var.asg_min}"
+  max_size             = "${var.asg_max}"
+  launch_configuration = "${aws_launch_configuration.app.id}"
+  target_group_arns    = ["${aws_alb_target_group.app.arn}"]
+}
+
+resource "aws_launch_configuration" "app" {
```

Comment:
I suspect our real world usage might require us to put the `aws_launch_configuration` in the parent caller, in that we'll have far too many params to pass through to this module, where as if you passed just the `aws_launch_configuration` in to this module you'd have less code that would be easier to follow